### PR TITLE
support listing taken replication slots [BF-2038]

### PR DESCRIPTION
**Pull request by ettanany**
### Proposed changes in this pull request

Support listing taken replication slots (useful to pre-check migration requirements when logical replication method is used).

### Type (put an `x` where ever applicable)
- [ ] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement
- [ ] Other

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [x] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable (in code and README.md).

### CODE REVIEW

File: aiven_db_migrate/migrate/pgmigrate.py [hunk 1]
```diff
@@ -348,6 +348,10 @@ def pg_roles(self) -> Dict[str, PGRole]:
     def replication_available(self) -> bool:
         return self.version >= "10"
 
+    @property
+    def taken_replication_slots(self) -> int:
```
Comment (Mulugruntz) [inline aiven_db_migrate/migrate/pgmigrate.py:352]: I don't understand this property name. Could we come up with some clearer name?

Comment (ettanany) [reply aiven_db_migrate/migrate/pgmigrate.py:352]: That's how we call a variable somewhere else in our code base. Basically what this returns is the number of replication slots that are already taken/occupied (I thought about "unavailable", "occupied", "used" and "taken" and at I went with "taken"), but I am open to your suggestions!

Comment (Mulugruntz) [reply aiven_db_migrate/migrate/pgmigrate.py:352]: > The `pg_replication_slots` view provides a listing of all replication slots that currently exist on the database cluster, along with their current state.

https://www.postgresql.org/docs/current/view-pg-replication-slots.html

So, how about `replication_slots_count` (concise) or `existing_replication_slots_count`? Also possible `num_replication_slots` or `total_replication_slots` (that one is a bit less clear that it's a number).

Yeah, while re-reading, I noticed that the name indicates a container of some sort, when we actually have an integer 😅. So I added `_count`.

Comment (ettanany) [reply aiven_db_migrate/migrate/pgmigrate.py]: :heavy_check_mark:

**Review by Mulugruntz — APPROVED** [review]

**Pull request merged**